src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/providers/okhttpMontaguAPIClient.kt

```diff
@@ -0,0 +1,129 @@
+package org.vaccineimpact.orderlyweb.security.providers
+
+import com.github.salomonbrys.kotson.fromJson
+import com.google.gson.JsonSyntaxException
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.slf4j.LoggerFactory
+import org.vaccineimpact.orderlyweb.Serializer
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.db.Config
+import java.io.IOException
+import java.security.cert.X509Certificate
+import javax.net.ssl.*
+
+
+interface MontaguAPIClient
+{
+    @Throws(MontaguAPIException::class)
+    fun getUserDetails(token: String): UserDetails
+
+    data class UserDetails(val email: String, val username: String, val name: String?)
+
+    // The following are identical to orderlyweb.models.Result, orderlyweb.models.ResultStatus,
+    // and orderlyweb.models.ErrorInfo but in principle they need not be and should either spec
+    // change could diverge, so defining Montagu specific models here
+    data class Result(val status: String, val data: Any?, val errors: List<ErrorInfo>)
+
+    data class ErrorInfo(val code: String, val message: String)
+    {
+        override fun toString(): String = message
+    }
+
+}
+
+class okhttpMontaguAPIClient(appConfig: Config = AppConfig()) : MontaguAPIClient
```

Comment:
The OkHttp library itself uses PascalCase for its class names (as do we normally for classnames!) so why is this lowercase?